[dnf5] Restructure microdnf advisory command to use subcommands and add output formatting

kontura opened the pull request.

kontura committed:
Add additional getters to advisories objects

kontura committed:
Add lookup_num to Pool class

This will be used atleast by advisories by getting buildtime

kontura committed:
Restructure microdnf advisory command to use subcommands

Conan-Kudo reviewed (approved):
Aside from the TODOs, this looks fine.

jrohel commented on libdnf/advisory/advisory.cpp:
```diff
@@ -100,6 +100,42 @@ std::string Advisory::get_severity() const {
+std::string Advisory::get_title() const {
+    // SOLVABLE_SUMMARY is misnamed, it actually stores the title
+    const char * title = get_pool(base).lookup_str(id.id, SOLVABLE_SUMMARY);
+    return title ? std::string(title) : std::string();
```
Instead of repeating the condition in several places, you can use our inline function `c_to_str` (from utils/string.hpp).

kontura replied:
Done

jrohel commented:
Swap the order of commits: 
First: "Add lookup_num to Pool class"
And then: "Add additional getters to advisories objects" which depends on previous one.

First: "Add output formatting for advisory list and info into libdnf-cli"
And then: "Restructure microdnf advisory command to use subcommands"  which depends on previous one.

jrohel commented:
1. It's not nice that the whole implementation of advisory outputs is in the header files - "libdnf-cli/output/advisoryinfo.hpp", "libdnf-cli/output/advisorylist.hpp".
2. These header files are used in the microdnf application, but are private. We have public headers in "include/".

The problem is with multiple header files. We will discuss at the meeting. Alternatively, I merge this and then reorganize multiple header files at once.

kontura committed:
Add output formatting for advisory list and info into libdnf-cli

kontura commented:
>     1. It's not nice that the whole implementation of advisory outputs is in the header files - "libdnf-cli/output/advisoryinfo.hpp", "libdnf-cli/output/advisorylist.hpp".
Ok, I have removed the templates because I think we won't need them anyway (and if we do we can add them back later), this allowed me to move all of the implementation into cpp files. 


>     2. These header files are used in the microdnf application, but are private. We have public headers in "include/".

I have added an `output` dir to `include/libdnf-cli/` and moved the headers there.
